### ISSUE TITLE
[vf] <apex:ouput* > considered unescaped in javascript even when escape="true"

### ISSUE THREAD

**Issue by yippie** [pmd/pmd#1091]
Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core].

Affects PMD Version:
6.2+

Rule: VfUnescapeEl

Description:
using <apex:outputText> in Javascript is flagged as unescaped when <apex:ouptuText> is escaped by default.

Code Sample demonstrating the issue:

```
<script>
   var options = {
      defaultSearchRecordType : '<apex:outputText value="{!defaultRecordTypeId}"/>'
    }
</script>
```

Running PMD through: CLI

**Comment by rsoesemann** [comment]
I was not able to reproduce this error using the code in the trunk. The rule indeed ignores tags that have "internal" escaping and there are test that check this. @yippie can you recheck that you are indeed using the latest PMD.

**Comment by rsoesemann** [comment]
Sorry but this code looks "nuts". Made up to break our fragile rules ;-)
What is the sense to have a string in Javascript code that contains a VF tag.

**Comment by jsotuyod** [comment]
Upon careful review, I don't think this is a FP at all.

`<apex:outputText>` will only perform XML escaping (replacing `<`, `>`, `&`, `"` and `'` with their corresponding XML entities), but that doesn't make it safe for inclusion on a JS context. For instance, on this example, if the rendered value ended in a backslash (`\`) `<apex:outputText>` wouldn't scape it, and produce:

```js
    var options = {
      defaultSearchRecordType : '…\'
    }
```

which would produce a syntax error preventing the execution of the rest of the code, and introducing unexpected results on the page.

The only way to properly avoid this (and what the rule actually validates) is using `JSENCODE` or `JSINHTMLENCODE` (assuming the XML escaping was actually desired to later embed the text in the HTML), as:

```js
    var options = {
      defaultSearchRecordType : '{!JSINHTMLENCODE(defaultRecordTypeId)}'
    }
```
